src/Sdk/DTPipelines/Pipelines/ObjectTemplating/PipelineTemplateConverter.cs

```diff
@@ -376,7 +389,26 @@ internal static Snapshot ConvertToJobSnapshotRequest(TemplateContext context, Te
                 return null;
             }
 
-            return new Snapshot(imageName);
+            return new Snapshot(imageName)
+            {
+                Condition = condition,
+                Version = version
+            };
+        }
+
+        private static bool IsSnapshotImageVersionValid(string versionString)
+        {
+            var versionSegments = versionString.Split(".");
+
+            if (versionSegments.Length != 2 ||
```

Comment:
so we only support A.B and not A.B.C for version?
This is different compare to `https://github.com/github/actions-larger-runners/issues/3170`

Reply:
Not yet, we want to start with <major>.* first and get feedback from customers. And then decide if we want to go beyond.